History Storage Overview Chart Fails to Refresh After Dataset Deletion

**Describe the bug**

In the Storage Dashboard, deleting a dataset from the "Top 10 Datasets by Size" chart does not trigger a visual refresh of the bar chart. While the backend action completes, the deleted dataset remains visible in the chart, and the selection card stays open.

This is confirmed by a [failing Selenium test](https://github.com/galaxyproject/galaxy/actions/runs/21668394001/job/62470089866), which times out waiting for the chart element to disappear/update.

```
FAILED lib/galaxy_test/selenium/test_history_storage.py::TestHistoryStorage::test_delete_dataset_from_storage_view - selenium.common.exceptions.TimeoutException: Message: Timeout waiting on CSS selector [[data-description='chart history top datasets by size'] rect[data-label='big']] to become absent.
```


**Screenshots**

https://github.com/user-attachments/assets/038cf27e-b85e-4c78-ad17-c2f5ff5fab5d

**Galaxy Version and/or server at which you observed the bug**

Galaxy Version: 26.1.dev0
Commit: a7c9eda928cd448dc69c7607f2df7809b4deb372

**Browser and Operating System**

Operating System: Windows, Linux, macOS
Browser: Firefox, Chrome, Chrome-based, Safari

**To Reproduce**

Steps to reproduce the behavior:
1. Navigate to User -> Storage Dashboard.
2. Go to History Storage Overview and locate the "Top 10 Datasets by Size" chart.
3. Click on a specific bar/dataset in the chart to open the info card.
4. Click the 'Delete' button and confirm the permanent deletion.
5. See the chart remains

**Expected behavior**

1. Chart Refresh: The chart should re-fetch or filter out the deleted dataset immediately upon confirmation.
2. UI Cleanup: The selection card (the "select card") for that dataset should close automatically.
3. Active/Deleted Sync: The "Active vs Deleted Total Size" chart below should also reflect the new storage distribution.

--- a/client/src/api/datasets.ts
+++ b/client/src/api/datasets.ts
@@ -100,8 +100,8 @@ export async function undeleteDataset(datasetId: string) {
     return data;
 }
 
-export async function deleteDataset(datasetId: string, purge: boolean = false) {
-    const { data, error } = await GalaxyApi().DELETE("/api/datasets/{dataset_id}", {
+export async function deleteDataset(datasetId: string, purge: boolean = false): Promise<void> {
+    const { error } = await GalaxyApi().DELETE("/api/datasets/{dataset_id}", {
         params: {
             path: { dataset_id: datasetId },
             query: { purge },
@@ -110,10 +110,9 @@ export async function deleteDataset(datasetId: string, purge: boolean = false) {
     if (error) {
         rethrowSimple(error);
     }
-    return data;
 }
 
-export async function purgeDataset(datasetId: string) {
+export async function purgeDataset(datasetId: string): Promise<void> {
     return deleteDataset(datasetId, true);
 }
 

--- a/client/src/components/User/DiskUsage/Visualizations/Charts/BarChart.vue
+++ b/client/src/components/User/DiskUsage/Visualizations/Charts/BarChart.vue
@@ -65,6 +65,7 @@ watch(
         props.valueFormatter,
     ],
     () => {
+        clearStaleSelection();
         // make sure v-if to conditionally display the div we're rendering this in
         // is available in the DOM before actually doing the rendering. Without
         // nextTick you cannot go from empty data -> chart when tweaking filtering
@@ -75,6 +76,16 @@ watch(
         });
     },
 );
+
+function clearStaleSelection() {
+    if (selectedDataPoint.value) {
+        const stillExists = props.data.some((d) => d.id === selectedDataPoint.value?.id);
+        if (!stillExists) {
+            selectedDataPoint.value = undefined;
+            emit("selection-changed", undefined);
+        }
+    }
+}
 
 function renderBarChart() {
     chartBars.value = drawChart();

--- a/client/src/components/User/DiskUsage/Visualizations/service.ts
+++ b/client/src/components/User/DiskUsage/Visualizations/service.ts
@@ -139,7 +139,6 @@ export async function undeleteDatasetById(datasetId: string): Promise<ItemSizeSu
     return data as unknown as ItemSizeSummary;
 }
 
-export async function purgeDatasetById(datasetId: string): Promise<PurgeableItemSizeSummary> {
-    const data = await purgeDataset(datasetId);
-    return data as unknown as PurgeableItemSizeSummary;
+export async function purgeDatasetById(datasetId: string): Promise<void> {
+    await purgeDataset(datasetId);
 }

--- a/client/src/components/User/DiskUsage/Visualizations/util.ts
+++ b/client/src/components/User/DiskUsage/Visualizations/util.ts
@@ -65,9 +65,9 @@ export function useDatasetsToDisplay() {
             return;
         }
         try {
-            const result = await purgeDatasetById(datasetId);
+            await purgeDatasetById(datasetId);
             const dataset = datasetsSizeSummaryMap.get(datasetId);
-            if (dataset && result) {
+            if (dataset) {
                 datasetsSizeSummaryMap.delete(datasetId);
                 successToast(localize("Dataset permanently deleted successfully."));
                 reloadData();

